flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -750,7 +751,28 @@ abstract class TableEnvironment(val config: TableConfig) {
     if (null == sinkTableName) throw TableException("Name of TableSink must not be null.")
     if (sinkTableName.isEmpty) throw TableException("Name of TableSink must not be empty.")
     if (!isRegistered(sinkTableName)) {
-      throw TableException(s"No table was registered under the name $sinkTableName.")
+      // try resolving and registering sink table from registered external catalogs
+      try {
+        val paths = sinkTableName.split("\\.")
+        if (paths.length > 1) {
```

Comment:
I am not exactly sure but this seems problematic to me. is there a guarantee that the first element of the path will always be `catalog` name? This also depends on whether the path requires to be fully-qualify or not, right?

Reply:
@walterddr thanks for reviewing. I assume full-path is required for both source table and sink table, as currently there is no "**USE** <db>" style of switching catalog/database supported. If fully qualified name is enforced, the first element of the path must be `catalog` name, right?